gui/src/installer/view.rs

```diff
@@ -1088,12 +1091,107 @@ pub fn help_backup<'a>() -> Element<'a, Message> {
     text(prompt::BACKUP_DESCRIPTOR_HELP).small().into()
 }
 
-pub fn define_bitcoin<'a>(
+pub fn define_bitcoin_node<'a>(
     progress: (usize, usize),
+    available_node_types: impl Iterator<Item = NodeType>,
+    selected_node_type: &NodeType,
+    node_view: Element<'a, Message>,
+    is_running: Option<&Result<(), Error>>,
+    can_try_ping: bool,
+    waiting_for_ping_result: bool,
+) -> Element<'a, Message> {
+    let col = Column::new()
+        .push(
+            available_node_types.fold(
+                Row::new()
+                    .push(text("Node type:").small().bold())
+                    .spacing(10),
+                |row, node_type| {
+                    row.push(radio(
+                        match node_type {
+                            NodeType::Bitcoind => "Bitcoin Core",
+                            NodeType::Electrum => "Electrum",
+                        },
+                        node_type,
+                        Some(*selected_node_type),
+                        |new_selection| {
+                            Message::DefineNode(message::DefineNode::NodeTypeSelected(
+                                new_selection,
+                            ))
+                        },
+                    ))
+                    .spacing(30)
+                    .align_items(Alignment::Center)
+                },
+            ),
+        )
+        .push(node_view)
+        .push_maybe(if waiting_for_ping_result {
+            Some(Container::new(
+                Row::new()
+                    .spacing(10)
+                    .align_items(Alignment::Center)
+                    .push(text("Checking connection...")),
+            ))
+        } else if is_running.is_some() {
+            is_running.map(|res| {
+                if res.is_ok() {
+                    Container::new(
+                        Row::new()
+                            .spacing(10)
+                            .align_items(Alignment::Center)
+                            .push(icon::circle_check_icon().style(color::GREEN))
+                            .push(text("Connection checked").style(color::GREEN)),
+                    )
+                } else {
+                    Container::new(
+                        Row::new()
+                            .spacing(10)
+                            .align_items(Alignment::Center)
+                            .push(icon::circle_cross_icon().style(color::RED))
+                            .push(text("Connection failed").style(color::RED)),
+                    )
+                }
+            })
+        } else {
+            Some(Container::new(Space::with_height(Length::Fixed(25.0))))
```

Comment:
nit (maybe depending on renderer/platform)
```suggestion
            Some(Container::new(Space::with_height(Length::Fixed(21.0))))
```